src/libexpr/attr-path.cc

```diff
@@ -67,7 +67,7 @@ std::pair<Value *, Pos> findAlongAttrPath(EvalState & state, const string & attr
 
         if (apType == apAttr) {
 
-            if (v->type != tAttrs)
+            if (v->normalType() != nAttrs)
```

Comment:
Instead of `normalType()` it would be nicer to write `v->isAttrs()`.

Do we actually need `normalType()`? I guess it's useful in `switch` statements...

Reply:
Other than it being useful in switch statements, it's also necessary in `ExprConcatStrings::eval`, which uses `firstType = vTmp.normalType()` to store the type, later comparing it to the types of other expressions. Similarly, `expectType` in `src/libexpr/flake/flake.cc` does a comparison between types. This wouldn't be possible with `is<Type>()` functions.